src/connection.h

```diff
@@ -0,0 +1,87 @@
+#ifndef __connection_h__
+#define __connection_h__ 1
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <netdb.h>  // For NI_MAXHOST/NI_MAXSERV
+#include <proton/engine.h>
+#include <proton/event.h>
+#include <proton/ssl.h>
+#include <qpid/dispatch/server.h>
+
+#include "server_private.h"
+
+#ifndef NI_MAXHOST
+#define NI_MAXHOST 1025
+#endif
+
+#ifndef NI_MAXSERV
+#define NI_MAXSERV 32
+#endif
+
+// Connection objects wrap Proton connection objects
+//
+// The `user_id` field is populated from the client SSL cert. See
+// ssl_uid_format in server.h for more info.
+struct qd_connection_t {
+    DEQ_LINKS(qd_connection_t);
+    char*                          name;
+    qd_server_t*                   server;
+    bool                           opened;  // An open callback was invoked for this connection
+    bool                           closed;
+    bool                           closed_locally;
+    int                            enqueued;
+    qd_timer_t*                    timer;  // Timer for initial setup
+    pn_connection_t*               pn_conn;
+    pn_session_t*                  pn_sessions[QD_SSN_CLASS_COUNT];
+    pn_ssl_t*                      ssl;
+    qd_listener_t*                 listener;
+    qd_connector_t*                connector;
```

Comment:
By and large the format convention for existing code is to have the indirection operator close to the variable name and not close to the type name. Is there a reason for changing this?
Attaching the operator to the variable name helps one to reason correctly about the variable's type when multiple variables are declared on the same line.

Reply:
Isn't that a pointer declaration, not an indirection operator?  As in, it would be an indirection operator if I were dereferencing something, but I'm not.

I have no problem flipping it, and ultimately I'll apply whatever style we decide should go in .clang-format.  I always think of it as type information that belongs with the type.

Reply:
I don't have a problem with either way. If you apply the new style in .clang-format will doing "clang-format -i src/*.c" change the style throughout the code? 
By that I mean, if we are changing the style here, we will have to do the same everywhere.

Reply:
"clang-format -i src/*.c would change all the .c files there, but I wouldn't make applying clang format an all-or-nothing thing.  I suggest using it when developing new code or doing a cleanup.

clang-format's formatting isn't perfect in every respect.  I think it's a net win, personally, but there are cases where it's not doing what you might want.  Pointer declarations is one example - with the pointer star aligned to the value, the [structs don't look right](https://stackoverflow.com/questions/38392889/clang-format-align-asterisk-of-pointer-declaration-with-variable-name).

Fortunately, you can apply it as needed and you can defeat the formatting if you need to with directives in comments.

Reply:
Works best with either `find` or `git`. Something like this to reformat all, `clang-format -i $(git ls-files -- *.c *.h)` and to reformat changed files, `clang-format -i $(git diff --name-only upstream/master..HEAD)`. Not sure if it is possible to only reformat changed sections of individual files like this.

Reply:
> Isn't that a pointer declaration, not an indirection operator?

It's both. **int p;** is an int declaration. **int * p;** is a pointer declaration. The difference between them is the **\***, the indirection operator. The operator has the same effect on declarations and on references.
It's unfortunate that the style settings don't have a way to specify the current style easily.

Reply:
Yeah, very unfortunate: https://github.com/llvm/llvm-project/blob/a8f720925599f8e44366438f1ccb4b4e9d9375ae/clang/lib/Format/WhitespaceManager.cpp#L641-L650